### PR TITLE
add option to control the drift linear gradient

### PR DESCRIPTION
**Pull request by samuelgarcia**
This:
* add a gradient minimum factor
* fix some memory blow due reading all templates from HDF5
* fix some memory blow due to acculating when padding and jittering templates

### CODE REVIEW

**Comment by alejoe91** [comment]
@samuelgarcia I think you need to add it in the recording_params as well

**Comment by samuelgarcia** [comment]
Where ?

**Comment by samuelgarcia** [comment]
I found it. In the yaml.

File: MEArec/default_params/recordings_params.yaml [hunk 1]
```diff
@@ -100,6 +100,7 @@ recordings:
   drift_mode_probe: 'rigid' # ['rigid', 'non-rigid']
   drift_fs: 100 # sampling frequency of drift signal in Hz
   non_rigid_gradient_mode: 'linear' # ['linear'] how the gradient on drift is applied on the prefered_dir. linear is 0->max velocity from up to bottom.
+  non_rigid_linear_min_factor: 0.4
```
Comment (alejoe91) [inline MEArec/default_params/recordings_params.yaml:103]:
```suggestion
  non_rigid_linear_min_factor: 0.5
```

**Comment by alejoe91** [comment]
@samuelgarcia can you push the changes from today?

**Comment by samuelgarcia** [comment]
> @samuelgarcia can you push the changes from today?

Oups. Done now.

**Review by alejoe91 — APPROVED** [review]

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**